packages/vm/src/vm.ts

```diff
@@ -127,8 +127,13 @@ export class VM {
     if (opts.evm) {
       this.evm = opts.evm
     } else {
-      const enableProfiler =
-        this._opts.profilerOpts?.reportAfterBlock ?? this._opts.profilerOpts?.reportAfterTx ?? false
+      let enableProfiler = false
+      if (
+        this._opts.profilerOpts?.reportAfterBlock === true ||
+        this._opts.profilerOpts?.reportAfterTx === true
```

Comment:
Aha! I just learned that I have interpreted `??` wrong, I thought it would continue the chain if the left statement was `false`. But it is not.